sql/core/src/main/scala/org/apache/spark/sql/RelationalGroupedDataset.scala

```diff
@@ -582,48 +582,59 @@ class RelationalGroupedDataset protected[sql](
   }
 
   /**
-   * Applies a vectorized python user-defined function to each cogrouped data.
-   * The user-defined function defines a transformation:
-   * `pandas.DataFrame`, `pandas.DataFrame` -> `pandas.DataFrame`.
-   *  For each group in the cogrouped data, all elements in the group are passed as a
-   * `pandas.DataFrame` and the results for all cogroups are combined into a new [[DataFrame]].
+   * Applies a vectorized python user-defined function to each cogrouped data. The user-defined
+   * function defines a transformation: `pandas.DataFrame`, `pandas.DataFrame` ->
+   * `pandas.DataFrame`. For each group in the cogrouped data, all elements in the group are
+   * passed as a `pandas.DataFrame` and the results for all cogroups are combined into a new
+   * [[DataFrame]].
    *
    * This function uses Apache Arrow as serialization format between Java executors and Python
    * workers.
    */
   private[sql] def flatMapCoGroupsInPandas(
-      r: RelationalGroupedDataset,
+      rs: Seq[RelationalGroupedDataset],
       expr: PythonUDF): DataFrame = {
     require(expr.evalType == PythonEvalType.SQL_COGROUPED_MAP_PANDAS_UDF,
       "Must pass a cogrouped map udf")
-    require(this.groupingExprs.length == r.groupingExprs.length,
-      "Cogroup keys must have same size: " +
-        s"${this.groupingExprs.length} != ${r.groupingExprs.length}")
+    val groupingExprLengthEquals = rs.map(_.groupingExprs.length).
+      forall(_ == this.groupingExprs.length)
+
+    require(groupingExprLengthEquals, s"Cogroup keys must have same size.")
```

Comment:
Could we keep the key sizes in the error message?

Something like `(this +: rs).map(_.groupingExprs.length).mkString(", ")`?

Reply:
Good suggestion, I will add them.